### PR TITLE
feat: wire CLI cost telemetry and vessel reports

### LINKED ISSUE #55: Wire cost telemetry and per-vessel cost reports into the CLI path
## Why this matters
The implementation plan makes this **Workstream 3 / Phase 1 / P0** alongside observability. xylem needs run-level token, latency, and spend data by default so later budget controls, routing decisions, and harness comparisons are evidence-based.

## Scope
- Record per-phase usage and aggregate it into per-vessel reports using `cli/internal/cost`.
- Persist machine-readable cost artifacts even when only partial usage data is available.
- Surface warnings when configured budget or cost thresholds are exceeded.
- Add role/model attribution so later routing and pruning decisions can be justified empirically.
- Ensure cost artifacts can be joined with traces and verification evidence.

## Acceptance checklist
- [ ] Every Claude/Copilot run emits cost data or an explicit `usage unavailable` reason.
- [ ] Operators can see per-vessel and per-phase cost/latency.
- [ ] Threshold warnings are surfaced in runner/status output.
- [ ] Tests cover usage aggregation, threshold warnings, and missing-usage edge cases.

## Dependencies
Pairs with #52 as the second half of Workstream 3.

## Roadmap alignment
Phase 1 -> Workstream 3

### PR DESCRIPTION
**Pull request by hnipps**
## Summary
- persist per-vessel cost reports and budget alert artifacts from the runner path even when usage is estimated or unavailable
- surface usage source, cost, and budget warning details in status output, reporter comments, and vessel observability attributes
- extend tests around usage aggregation, threshold warnings, and unavailable-usage edge cases

Closes https://github.com/nicholls-inc/xylem/issues/55

### CODE REVIEW

**Comment by hnipps** [comment]
**xylem — phase `analyze` completed** (1m1.068328s)

_Usage:_ $0.0000, 2023

<details>
<summary>Phase output (click to expand)</summary>

I’m pulling the PR state and review metadata first so I can classify each thread against the current branch contents.

`gh pr checkout` hit a missing remote-ref error, so I’m checking the current git state and PR metadata directly, then I’ll fall back to checking out the PR head by ref.



</details>

**Comment by hnipps** [comment]
**xylem — phase `check_deps` completed** (1m45.907669s)

_Usage:_ $0.0000, 1789

<details>
<summary>Phase output (click to expand)</summary>

Running gh to fetch PR labels for PR #248 to determine whether this PR is part of the harness-impl wave. Also reporting intent: "Checking PR labels". I'll run gh pr view and parse labels.



</details>

**Comment by hnipps** [comment]
**xylem — all phases completed**

| Phase | Duration | Cost | Tokens | Status |
|-------|----------|------|--------|--------|
| check_deps | 1m45.907669s | $0.0000 | 1789 | completed |

Total: 1m45.907669s — $0.0000, 1789 tokens (estimated)

**Comment by hnipps** [comment]
**xylem — phase `respond` completed** (2m52.488576s)

_Usage:_ $0.0000, 1827

<details>
<summary>Phase output (click to expand)</summary>

I found one actionable review thread: the vessel summary can disagree with the cost report when usage is recorded only via the tracker. I’m inspecting the summary/cost-report path and then I’ll patch it so status, reporter, and observability stay aligned.

I’ve confirmed the gap: `buildCostReport` already treats tracker-only records as estimated usage, but `buildSummary` still ignores them. I’m wiring the summary to use the same tracker view and adding a regression test for the tracker-only path.

The fix is in: summaries now derive usage metadata from the tracker when tracker data exists, so tracker-only runs no longer show `not_applicable` while the cost report shows estimated usage. I’m running the repo checks again and then I’ll package the reviewer reply text.



</details>

File: cli/internal/runner/summary.go [hunk 1]
```diff
@@ -197,9 +203,13 @@ func (s *vesselRunState) buildSummary(state string, endedAt time.Time) *VesselSu
...
 	summary.TotalCostUSDEst += s.extraCostUSDEst
 	summary.TotalTokensEst = summary.TotalInputTokensEst + summary.TotalOutputTokensEst
+	summary.UsageSource, summary.UsageUnavailableReason = summarizeUsageSource(summary.Phases, summary.TotalTokensEst, summary.TotalCostUSDEst)
 
 	if s.costTracker != nil {
 		summary.BudgetExceeded = s.costTracker.BudgetExceeded()
+		alerts := s.costTracker.Alerts()
+		summary.BudgetAlertCount = len(alerts)
+		summary.BudgetWarning = hasBudgetWarning(alerts)
 	}
```
[the remark this answers is not in the document]
Comment (hnipps) [reply cli/internal/runner/summary.go:213]: Thanks — fixed. Summaries now derive usage metadata from the cost tracker when tracker records exist (tracker-only runs will show "estimated" usage). Added a regression test (TestBuildSummaryUsesTrackerTotalsWhenPhaseTotalsAreUnavailable) and updated persistRunArtifacts to save aligned artifacts. Pushed changes in commit: fix: address review feedback on #248

**Comment by hnipps** [comment]
**xylem — phase `push` completed** (3m35.107143s)

_Usage:_ $0.0000, 1924

<details>
<summary>Phase output (click to expand)</summary>

Searching the codebase for the summary/cost-report functions and tracker usage to locate the source of the bug. Running parallel grep queries plus reporting intent.



</details>

**Comment by hnipps** [comment]
**xylem — all phases completed**

| Phase | Duration | Cost | Tokens | Status |
|-------|----------|------|--------|--------|
| analyze | 1m1.068328s | $0.0000 | 2023 | completed |
| respond | 2m52.488576s | $0.0000 | 1827 | completed |
| push | 3m35.107143s | $0.0000 | 1924 | completed |

Total: 7m28.664047s — $0.0000, 5774 tokens (estimated)

### Verification evidence

| Claim | Level | Checker | Result |
|-------|-------|---------|--------|
| Gate passed for phase "respond" | untyped | cd cli && go vet ./... && go build ./cmd/xylem && go test ./... | :white_check_mark: |

<details>
<summary>Trust boundaries</summary>

- **Gate passed for phase "respond"** — No trust boundary declared

</details>
